### DISCUSSION TITLE
How can I connect to a postgres server running on my host from budibase running on a docker compose?

### DISCUSSION THREAD

**Discussion by cverdes** [Budibase/budibase#6517] [Help] [answered]
Dear Budibase developers,

I running a postgres server un a ubuntu 20.04 machine that has budibase installed and running within a docker composer.

The ip of the host machine is 192.168.0.104 and postgres is listening on port 5432

I tried this connection configuration:
![image](https://user-images.githubusercontent.com/6440942/176784653-6a261259-5531-4a73-b4bc-6bbbf96eedd6.png)

But I get this error:
![image](https://user-images.githubusercontent.com/6440942/176785148-6f683032-ab53-4560-b41a-3e01710901a3.png)

I don't know much about docker. Please could you explain me what am I doing wrong or how to configure docker to connect postgres server from budibase running with docker-compose up?

thanks in advance for your help

**Comment by melohagan** [comment]
Hi @cverdes 

So Budibase is installed on the same machine as the Postgres server?

Have you tried replacing the host with `host.docker.internal` ?

**Reply by cesarverdesslots** [reply]
Dear @melohagan,

Yes, Budibase is installed (running into docker) on the same machine (running Ubuntu 20.04) as the Postgres server (the postgres server was installed with `sudo apt-get install postgresql`) and it's running ok because I can login with pgadmin4 without problem:

![image](https://user-images.githubusercontent.com/83496099/176897210-6ced55ab-88f5-47d5-a146-6226cf77ebb6.png)


When I try with `host.docker.internal` I get the following error:

![image](https://user-images.githubusercontent.com/83496099/176897612-a06f7253-fff1-4272-89ba-cf84fbc3e071.png)

**Comment by melohagan** [comment, marked as answer]
Hi @cesarverdesslots Apologies, I forgot linux is different. Try: `172.17.0.1` for your host.

https://docs.budibase.com/docs/build-a-crud-app-sql#connecting-to-your-sql-tables

**Reply by FlyingNimbus** [reply]
This also worked for me, but I don't understand it. **Would you mind explaining how this works so we can learn from it?** I assumed 127.0.0.1 or the IP of the postgres container would work, but I don't see how 172.17.0.1 works.

(I'm running docker on Windows 10 with 1 container for budibase and 1 container for postgres)

**Reply by melohagan** [reply]
Hey @FlyingNimbus 

This is specifically relating to docker when you are running locally. When you run Budibase locally inside a docker container and you have a database running on localhost then you need to tell docker the route to the localhost of the machine.

You do this with `host.docker.internal` on Windows and MacOS, and `172.17.0.1` on Linux. 

There's an article here that may be useful: https://dev.to/natterstefan/docker-tip-how-to-get-host-s-ip-address-inside-a-docker-container-5anh

**Comment by cesarverdesslots** [comment]
Thank you very much @melohagan!

When I set 172.17.0.1 as host connection but I got the error `no pg_hba.conf entry for host "172.18.0.8", user "postgres", database "postgres", SSL encryption`

I add the instruction `host  all  all  all  md5` whitin the `pg_hba.conf` file and reset the service with `sudo systemctl restart postgresql.service` and IT WORKED!

Thanks again and sorry for the inconvenience!

:+1:

**Reply by melohagan** [reply]
No bother at all!

**Reply by ganaranjos** [reply]
Hello Guys. 
I have the same issue, just the difference is that I´m using Ubuntu 20.04 with terminal (AWS machine). Considering some changes in pg_hba.conf......how can I acces to this path to edit by terminal? I´m not an expert in Postgres DB.
